src/app/dashboard/competitions/[competitionId]/register-form/create/components/CreateCompetitionRegisterForm.tsx

```diff
@@ -0,0 +1,217 @@
+    undefined,
+  );
+  const [newFieldName, setNewFieldName] = useState<string[]>([]);
+  const router = useRouter();
+  const [newFieldType, setNewFieldType] = useState<string[]>([]);
+  const [currentFieldName, setCurrentFieldName] = useState<string>('');
+  const [currentFieldType, setCurrentFieldType] = useState<string>('');
+  const [form, fields] = useForm({
+    lastResult,
+    onValidate({formData}) {
+      return parseWithZod(formData, {schema: CompetitionRegisterFormSchema});
+    },
+    shouldValidate: 'onBlur',
+    shouldRevalidate: 'onInput',
+  });
+
+  useEffect(() => {
+    if (lastResult?.status === 'success') {
+      toast.success('Formulario de registro creado correctamente');
+      router.push(`/dashboard/competitions/${competition.id}`);
+    } else if (lastResult?.status === 'error') {
+      toast.error(lastResult.error?.message?.[0] || 'Error al crear el formulario de registro');
+    }
+  }, [lastResult, router, competition.id]);
+
+  function handleAddField() {
+    if (currentFieldName && currentFieldType) {
+      setNewFieldName([...newFieldName, currentFieldName]);
+      setNewFieldType([...newFieldType, currentFieldType]);
+      setCurrentFieldName('');
+      setCurrentFieldType('');
+    }
+  }
+
+  function handleDeleteField(index: number) {
+    setNewFieldName(newFieldName.filter((_, i) => i !== index));
+    setNewFieldType(newFieldType.filter((_, i) => i !== index));
+  }
+
+  return (
+    <div className="card bg-base-100 shadow-xl">
+      <div className="mt-4">
+        <GoBackButton />
+      </div>
+      <div className="card-body">
+        <h1 className="card-title mb-6 text-2xl">Crear formulario de registro para {competition.name}</h1>
+
+        <form id={form.id} onSubmit={form.onSubmit} action={formAction} className="space-y-4">
+          <FormInput
+            label="Título"
+            name={fields.title.name}
+            type="text"
+            required
+            placeholder="Título del formulario de registro"
+            errors={fields.title.errors}
+          />
+
+          <FormInput
+            label="Descripción"
+            name={fields.description.name}
+            type="text"
+            required
+            placeholder="Descripción del formulario de registro"
+            errors={fields.description.errors}
+          />
+
+          <FormCheckbox
+            label="Gratuito"
+            name={fields.isFree.name}
+            checked={isFreeChecked}
+            onChange={() => setIsFreeChecked(!isFreeChecked)}
+            errors={fields.isFree.errors}
+          />
+
+          {isFreeChecked === false && (
+            <>
+              <PriceInput
+                name={fields.price.name}
+                errors={fields.price.errors}
+              />  
+              <br></br>
+              <Toogle
+                label={isPaymentToggleChecked ? "Datos de pago" : "Link de pago"}
```

Comment:
Lo primero, recomiendo cambiar `Datos de pago` por `Datos de transferencia`.

Lo segundo, y más importante recomiendo hacer un cambio en el orden del toggle. A nivel de UX se debe mostrar la opción a la que cambiará si es que se selecciona, no la opción en la que se está actualemente.


```suggestion
                label={isPaymentToggleChecked ? "Link de pago" : "Datos de transferencia"}
```